### PR TITLE
Add Criterion Benchmarks

### PR DESCRIPTION
**Pull request by tyurek**
This PR addresses issues #30 and #33 

It creates a rust-stable-compatible end-to-end benchmarking example which can be called via `cargo bench`.
The example measures the performance of the Presign, Keygen, and Auxinfo protocols with 3, 10, and 20 nodes.

Wrt issue #30, this example demonstrates that the currently exposed interface is sufficient for an external caller and does not appear to need modificiation

### CODE REVIEW

File: src/participant.rs
Comment (tyurek) [reply src/participant.rs:182]: This benchmark, like the new tests, shuffles the message arrival order. In doing so, this print statement gets hit multiple times per run, but hitting it isn't an error. (I'm not using it as only a safeguard. The code that calls this is intended to attempt to run some functions multiple times. This just prevents anything from happening after the first)

This line is probably useful as a debugging tool, but currently it prints multiple times per error-free run so I took it out.

**Comment by tyurek** [comment]
good point. This update should clean it up a bit more

File: benches/my_benchmark.rs
Comment (gatoWololo) [inline benches/my_benchmark.rs:13]: I don't really understand how `&mut *inboxes` works. I see the compiler suggests doing it this way :sweat_smile:

Can we do it this way:
```diff
+ for (id, inbox) in inboxes.into_iter() {
+     if *id == message.to() {
- for (&id, inbox) in &mut *inboxes {
-     if id == message.to() {
```
I think it makes it a little more obvious what we are doing.

**Review by gatoWololo — APPROVED** [review]
Looks good! Just two comments:
1) Change the way we iter through inboxes (see comment below).
2) Can you squash the commits into a single one before merging?

I approved the PR so you won't need me to re-approve it once you make the changes.

**Pull request merged**
